core/src/test/java/org/apache/iceberg/TestMetadataTableFilters.java

```diff
@@ -129,14 +137,28 @@ private int expectedScanTaskCount(int partitions) {
         }
       case DATA_FILES:
       case DELETE_FILES:
+      case ALL_DELETE_FILES:
         return partitions;
       case ALL_DATA_FILES:
         return partitions * 2; // ScanTask for Data Manifest in DELETED and ADDED states
+      case ALL_FILES:
+        if (formatVersion == 1) {
+          return partitions * 2; // ScanTask for Data Manifest in DELETED and ADDED states
+        } else {
+          return partitions * 4; // ScanTask for Delete and Data File in DELETED and ADDED states
+        }
       default:
         throw new IllegalArgumentException("Unsupported metadata table type:" + type);
     }
   }
 
+  private boolean allFileTableTest(MetadataTableType tableType) {
+    return Sets.newHashSet(MetadataTableType.ALL_DATA_FILES,
+            MetadataTableType.ALL_DATA_FILES,
+            MetadataTableType.ALL_FILES)
```

Comment:
Nit: Consider making this a `static final` constant.

Reply:
+1 for a constant